compiler/rustc_passes/src/dead.rs

```diff
@@ -434,12 +434,13 @@ impl<'tcx> Visitor<'tcx> for MarkSymbolVisitor<'tcx> {
         let tcx = self.tcx;
         let has_repr_c = self.repr_has_repr_c;
         let has_repr_simd = self.repr_has_repr_simd;
+        let effective_visibilities = &tcx.effective_visibilities(());
         let live_fields = def.fields().iter().filter_map(|f| {
             let def_id = f.def_id;
             if has_repr_c || (f.is_positional() && has_repr_simd) {
                 return Some(def_id);
             }
-            if !tcx.visibility(f.hir_id.owner.def_id).is_public() {
+            if !effective_visibilities.is_reachable(f.hir_id.owner.def_id) {
                 return None;
             }
             if tcx.visibility(def_id).is_public() { Some(def_id) } else { None }
```

Comment:
Should this use effective_visibility too?

Reply:
Hi!
Thanks for the quick review!
My reasoning was that if the "owner" of *Id* is reachable, it is enough for *Id* to be public to be reachable as well. Hence I kept `visibility` here, maybe under the semi-conscious assumption that it was "cheaper" to computer than reachability. I am not too familiar with the compiler structure and what is or not costly so if you prefer to have reachability at both places let me know and I'll change.

Reply:
The slow part is computing effective visibilities, but they are unconditionally computed once for the crate, so this does not matter. Once they are computed, it's just a matter of reading an array. So using effective visibility here should be (negligibly) faster than querying the nominal visibility.